### PR TITLE
fixes #2764 - settings defined in settings.yaml file are now read-only

### PR DESCRIPTION
**Pull request by domcleal**
@ohadlevy replaces #854

### CODE REVIEW

**Comment by lzap** [comment]
Do it now, refactor later. +1

**Comment by lzap** [comment]
_This pull request is considered as blocker for Foreman 1.3 release._

**Comment by lzap** [comment]
Please someone review that since @ohadlevy is out today! Thanks!

File: app/models/setting.rb [hunk 1]
```diff
@@ -139,24 +140,31 @@ def parse_string_value val
...
-      s.update_attribute(:default, opts[:default])
-      s
+      create_existing(s, opts)
     end
   end
 
   private
 
+  def self.create_existing(s, opts)
+    s.instance_variable_set("@readonly", false) if (saved_readonly_state = s.readonly?)
```
Comment (ares) [inline app/models/setting.rb]: maybe adding `readwrite!` would be a bit nicer or make a method accepting block like `s.bypass_readonly { |s| s.update }`

Comment (domcleal) [inline app/models/setting.rb]: I've added the block method, I like that!

**Comment by ares** [comment]
:+1:

**Comment by domcleal** [comment]
I seem to have broken the world, let me investigate.

**Comment by skottler** [comment]
The code looks good and it works in my tests. Waiting for tests to run and I'll merge assuming they pass.

**Comment by domcleal** [comment]
Closing in favour of #879, where @ares has kindly fixed the test issue.

**Pull request closed without merging**